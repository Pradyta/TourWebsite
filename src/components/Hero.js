import React, { useContext } from "react";
import "./cssFiles/HeroStyles.css";
import { AiOutlineSearch } from "react-icons/ai";
import Video from "../assets/maldivesVideo.mp4";
import { searchLink } from "../data";
import SearchBar from "./SearchBar";

function Hero() {
  return (
    <div className="hero">
      <video autoPlay loop muted id="video">
        <source src={Video} type="video/mp4" />
      </video>
      <div className="content home">
        <span className="text1">Your Journey Your Story</span>
        <span className="text1">Choose Your favourite Destination</span>

        <div>
          <SearchBar
            placeholder="Enter Destination Here ..."
            data={searchLink}
          />
        </div>
      </div>
    </div>
  );
}

export default Hero;
