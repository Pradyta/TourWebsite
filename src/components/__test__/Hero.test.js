import React from "react";
import { render, screen } from '@testing-library/react';
import Hero from "../Hero";

describe("Hero Component", () => {
  test("should render Hero compoennt", () => {
    const {asFragment} = render( 
      <Hero />
  );
  expect(asFragment(<Hero/>)).toMatchSnapshot();

  });

    test("header renders with react testing tutorial in the document", () => {
      const component = render( 
        <Hero />
    );
    const linkElement = component.getByText(/Your Journey Your Story/i);
    expect(linkElement).toBeInTheDocument();
  });
});
