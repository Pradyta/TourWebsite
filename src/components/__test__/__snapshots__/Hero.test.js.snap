// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component should render Hero compoennt 1`] = `
<DocumentFragment>
  <div
    class="hero"
  >
    <video
      autoplay=""
      id="video"
      loop=""
    >
      <source
        src="test-file-stub"
        type="video/mp4"
      />
    </video>
    <div
      class="content"
    >
      <h1>
        Your Journey Your Story
      </h1>
      <h2>
        Choose Your favourite Desination
      </h2>
      <div>
        <div
          class="search"
        >
          <div
            class="searchInputs"
          >
            <input
              placeholder="Enter Destination Here ..."
              type="text"
              value=""
            />
            <div
              class="searchIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
